### PR TITLE
Switch version from main to fix version.

### LINKED ISSUE #52: Version defaults to main which can be unstable
**Is your feature request related to a problem? Please describe.**
The main branch of HDM is often in an unstable state or broken state and it can be confusing to new users when using the modules default for version

**Describe the solution you'd like**
A default version of latest HDM stable version

**Describe alternatives you've considered**
Maybe just documenting it clearly in the readme would help?

### PR DESCRIPTION
**Pull request by tuxmea**
Users must check the version to use from hdm website. Updated documentation

fixes #52

### CODE REVIEW

**Review by rwaffen — COMMENTED** [review]
LGTM, but not sure about the doc/* files

File: doc/_index.html [hunk 1]
```diff
@@ -0,0 +1,163 @@
+<!DOCTYPE html>
```
Comment (bastelfreak) [inline doc/_index.html:1]: this looks like puppet-strings docs as html output. We only save the markdown version (REFERENCE.md). Please delte `doc/` and update the REFERENCE.md with `bundle exec rake reference`

File: manifests/init.pp [hunk 2]
```diff
@@ -119,9 +120,9 @@
 #   include hdm
 class hdm (
   # installation parameter
+  String[1]                     $version,
   Enum['docker', 'rvm']         $method                = 'docker',
   Boolean                       $manage_docker         = true,
-  String[1]                     $version               = 'main',
```
Comment (bastelfreak) [inline manifests/init.pp:124]: can't we set `1.0.1` as default version? Our goal is always to provide modules that work with `include $module` and will result in a working service with sane defaults.

Comment (tuxmea) [reply manifests/init.pp:124]: Setting this to 1.0.1 means, that we must update puppet-hdm after every hdm release. Besides this: versions differ between docker and rvm setup.

Comment (rwaffen) [reply manifests/init.pp:124]: was thinking the same 🤔

File: doc/_index.html (hunk 1 above)
Comment (tuxmea) [reply doc/_index.html]: Done

File: manifests/init.pp (hunk 2 above)
Comment (tuxmea) [reply manifests/init.pp:124]: Version has been set to 1.0.1

**Review by rwaffen — APPROVED** [review]

**Review by bastelfreak — APPROVED** [review]

**Comment by bastelfreak** [comment]
Will this break existing setups when the module switches their version from main to 1.0.1? Then we need to add the backwards-incompatible label to the PR.

**Comment by rwaffen** [comment]
i dont think so. main and 1.0.1 more or less are the same. main has two merges more, one from dependabot and one for deep-merging if datadir is missing in hiera.yaml.

**Comment by davidsandilands** [comment]
Looks good to me for issue I raised 👍

File: manifests/init.pp [hunk 3]
```diff
@@ -119,9 +120,9 @@
 #   include hdm
 class hdm (
   # installation parameter
+  String[1]                     $version               = '1.0.1',
```
Comment (smortex) [inline manifests/init.pp:123]: Each change here will require a new major version. Unless this "hdm" is something that is unlikely to evolve, it is probably better to **not** set a default value and put that responsibility on the end user.

File: README.md [hunk 4]
```diff
@@ -69,12 +69,25 @@ If you want to install HDM using RVM, you need the following modules:
...
+```
+
+Another option is to use class resource type declaration:
+
+```puppet
+class { 'hdm':
+  version => '1.0.1',
+}
+```
+
```
Comment (smortex) [inline README.md:90]: I would drop Hiera stuff completely:

    The most simple approach is to just include the hdm class and provide the version parameter:
    
    ```puppet
    class { 'hdm':
      version => '1.0.1',
    }
    ```
    
    You can find the released versions on [HDM releases page](https://github.com/betadots/hdm/releases)

People who are used to to this old-style pattern with Hiera know how to "convert", and those who don't won't be confused ("include" in the sentense is probably not the best, but i am fine with it and can't figure out something concise and suited).

Comment (smortex) [reply README.md:90]: Ah, lol, it's a hiera data visualization tool :facepalm:. Maybe it make sense in that case :shrug:

File: manifests/init.pp (hunk 3 above)
Comment (bastelfreak) [reply manifests/init.pp:123]: I'm not sure about this. If a user want's to run the latest release, they need to update their local code/hiera. It doesn't matter if the module has no version vs an old version. But with a default version in the module the user can get started with `include hdm` and we've that pattern in most other modules, for example puppet/prometheus. We don't update the modules to always ship the latest release.

**Pull request merged**